core/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -409,6 +409,52 @@ class Router(formatOps: FormatOps) {
                 .withIndent(2, expire, Left)
             )
         }
+
+      // Parameter opening for one parameter group. This format works
+      // on a group-by-group basis.
+      case FormatToken(open @ LeftParen(), right, between)
+          if style.verticalMultiline && isDefnSite(leftOwner) =>
+        val close = matchingParentheses(hash(open))
+        val indentParam = Num(4)
+        val indentSep = Num(2)
+        if (isTuple(leftOwner)) {
+          Seq(
+            Split(NoSplit, 0).withPolicy(
+              SingleLineBlock(close, disallowInlineComments = false))
+          )
+        } else {
+          val oneLinePerArgPolicy =
+            OneArgOneLineSplit(
+              open,
+              noTrailingCommas = style.poorMansTrailingCommasInConfigStyle)
+
+          // A `paramSep` is a `)(`, aka the seperator between two parameter
+          // groups. Prepend a newline to these.
+          val noNewLineOnParamSep: PartialFunction[Decision, Decision] = {
+            case Decision(t @ FormatToken(_, `close`, _), _) =>
+              Decision(t,
+                Seq(Split(Newline, 0)
+                  .withIndent(indentSep, close, Right)
+                )
+              )
+          }
+          val configStyleLike =
+            oneLinePerArgPolicy.andThen(noNewLineOnParamSep)
+          // Find the very first leftParen in the `def`. This will tell us
+          // if we are in the first group or not.
+          val isFirstGroup =
+            leftOwner.tokens.find {
+              case LeftParen() => true
+              case _ => false
+            }.filter(_ == open).isDefined
+          val mod = if (isFirstGroup) Newline else NoSplit
+          Seq(
+            Split(mod, 0)
```

Comment:
That would cause the split to happen partly. So it won't split on the first parameter group but it would on the second (if not splitting it causes it to go over the maxColumn). 

The test shows:

```scala
def format_![T <: Tree](code: String, foo: Int)(f: A => B
  )(implicit ev: Parse[T], ev2: EC): String
```

I need to somehow know, at the first `(` whether maxColumn has been violated or not and do one style or the other for the whole `def`.